adagucserverEC/CDBAdapterPostgreSQL.cpp

```diff
@@ -1080,3 +1084,86 @@ int CDBAdapterPostgreSQL::addFilesToDataBase() {
 #endif
   return 0;
 }
+
+int CDBAdapterPostgreSQL::storeLayerMetadata(const char *datasetName, const char *layerName, const char *metadataKey, const char *metadataBlob) {
+#ifdef MEASURETIME
+  StopWatch_Stop(">CDBAdapterPostgreSQL::storeLayerMetadata");
+#endif
+  CPGSQLDB *dataBaseConnection = getDataBaseConnection();
+  if (dataBaseConnection == NULL) {
+    return -1;
+  }
+
+  CT::string tableColumns("datasetname varchar (255), layername varchar (255), metadatakey varchar (255), updatetime TIMESTAMPTZ, blob JSONB, PRIMARY KEY (datasetname, layername, metadatakey)");
```

Comment:
Two things:
- We use a mix of 255 and 511 length voor `varchar`. Why did you pick 255 here?
- In other places I see `UNIQUE`, not `PRIMARY KEY`. Looking at the [docs](https://www.postgresql.org/docs/current/sql-createtable.html), I think `PRIMARY KEY` is better here. But not clear why it is not used in other places. But not a big deal.

Reply:
Lets keep `PRIMARY KEY` here. I would avoid changing the others.



Changed the varchar to 511 for datasetname and layername, although 255 is probably plenty.